### PR TITLE
Add `Dynamic Text Template` infrastructure.

### PR DESCRIPTION
**Pull request by maliming**
Breaking change: `LocalizationResource(Type)` of `TemplateDefinition` class is changed to `LocalizationResourceName(string)`

### CODE REVIEW

File: framework/src/Volo.Abp.TextTemplating.Core/Volo/Abp/TextTemplating/ITemplateDefinitionManager.cs [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
 {
     [NotNull]
-    TemplateDefinition Get([NotNull] string name);
+    Task<TemplateDefinition> GetAsync([NotNull] string name);
 
     [NotNull]
-    IReadOnlyList<TemplateDefinition> GetAll();
+    Task<IReadOnlyList<TemplateDefinition>> GetAllAsync();
 
     [CanBeNull]
```
Comment (hikalkan) [inline framework/src/Volo.Abp.TextTemplating.Core/Volo/Abp/TextTemplating/ITemplateDefinitionManager.cs:15]:
```suggestion
    [ItemCanBeNull]
```

File: framework/src/Volo.Abp.TextTemplating.Core/Volo/Abp/TextTemplating/ITemplateDefinitionManager.cs [hunk 2]
```diff
@@ -1,16 +1,17 @@
 ﻿using System.Collections.Generic;
+using System.Threading.Tasks;
 using JetBrains.Annotations;
 
 namespace Volo.Abp.TextTemplating;
 
 public interface ITemplateDefinitionManager
 {
     [NotNull]
```
Comment (hikalkan) [inline framework/src/Volo.Abp.TextTemplating.Core/Volo/Abp/TextTemplating/ITemplateDefinitionManager.cs:9]:
```suggestion
    [ItemNotNull]
```

File: framework/src/Volo.Abp.TextTemplating.Core/Volo/Abp/TextTemplating/ITemplateDefinitionManager.cs [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
-    TemplateDefinition Get([NotNull] string name);
+    Task<TemplateDefinition> GetAsync([NotNull] string name);
 
     [NotNull]
```
Comment (hikalkan) [inline framework/src/Volo.Abp.TextTemplating.Core/Volo/Abp/TextTemplating/ITemplateDefinitionManager.cs:12]:
```suggestion
    [ItemNotNull]
```

**Review by hikalkan — APPROVED** [review]

**Comment by hikalkan** [comment]
I will do the changes, no problem. Can you please create a migration guide and write that change for 7.4.

**Comment by maliming** [comment]
> I will do the changes, no problem. Can you please create a migration guide and write that change for 7.4.

ok.

**Pull request merged**